doc/obc-provisioner.md

```diff
@@ -78,6 +78,29 @@ spec:
     bucketclass: custom-bucket-class
 ```
 
+If the requested BucketClass is in namespace other than NooBaa system namespace then the BucketClass namespace can be specified via `spec.additionalConfig.bucketclassnamespace`.
+
+Example:
+
+```bash
+TODO
+```
+
+```yaml
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucketClaim
+metadata:
+  name: my-bucket-claim
+  namespace: my-app
+spec:
+  generateBucketName: my-bucket
+  storageClassName: noobaa.noobaa.io
+  additionalConfig:
+    bucketclass: custom-bucket-class
+    bucketclassNamespace: test # BucketClass will be searched in this Namespace
```

Comment:
I think that OBC can only access bucketclass in its own NS or noobaa NS. in that case, `bucketclassNamespace` is only necessary if there are 2 bucketclasses with the same name in both namesapces. can we just have a priority in that case?
@utkarsh-pro @romayalon @guymguym WDYT?

Reply:
@dannyzaken Sounds ok to me. So the bucket class lookup will be from the obc namespace first, and if not found will fallback to looking up the bc name in the noobaa namespace?

Reply:
@dannyzaken lemme know if I am getting this right.

Right now (Noobaa 5.11), an OBC can be created in any namespace and it can refer a bucketclass which is (and can be only in) Noobaa System's namespaces.

With this PR, `BucketClass` can also be in any namespace, so is the suggestion that instead of allowing OBC to refer BucketClass of any name in any namespace (current, i.e. in PR implementation) restrict OBC to refer to BucketClass which are either present in OBC's namespace (app-namespace) or is in Noobaa system's namespace?

Reply:
Addressed